pkg/asset/installconfig/gcp/validation.go

```diff
@@ -66,8 +67,36 @@ func ValidateInstanceType(client API, ic *types.InstallConfig, fieldPath *field.
 			errMsg := fmt.Sprintf("instance type does not meet minimum resource requirements of %d MB Memory", req.minimumMemory)
 			allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, errMsg))
 		}
+	} else if index := strings.Index(instanceType, "custom-"); index != -1 {
+
+		// Ensure the CPU and MEMORY are defined in custom instance type names
+		if customConfig := strings.Split(instanceType[index+len("custom-"):], "-"); len(customConfig) == 2 {
```

Comment:
How would you feel about using a regular expression rather than `strings.Index` and `strings.Split`? The regular expression makes it a bit more concise and (arguably) conveys the expected format for custom instance types more clearly. I am fine if you want to keep it as is, too.
```
regexp.MustCompile(`^(\w+-)?custom-(\d+)-(\d+)$`)
```

Reply:
I think we should maybe request the instance type from GCP api instead of this string-extraction. ServiceDelivery previously mentioned how these can change.

Reply:
Switched to using the GCP API to reduce future maintenance on this code.